### PR TITLE
Exception handling

### PR DESCRIPTION
**Pull request by janahofmann**
- unit tests for exceptions+fences and nested speculation
- added files that are generated by running tests to gitignore

### CODE REVIEW

**Comment by janahofmann** [comment]
@microsoft-github-policy-service agree [company="microsoft"]

**Comment by janahofmann** [comment]
@microsoft-github-policy-service agree company="Microsoft"

**Comment by OleksiiOleksenko** [comment]
sorry for the delay, I'll look into it later today

File: src/x86/tests/unit_model.py [hunk 1]
```diff
@@ -364,6 +467,73 @@ def test_ct_meltdown(self):
...
+        ctraces = self.get_traces(model, ASM_FAULTY_ACCESS, [input_], nesting=2)
+        expected_trace = hash(tuple([
+            cbase, mbase + 4096,  # fault
+            cbase, mbase + 4096,  # speculative injection
+            cbase + 4, mbase + 4097, # second fault
+            # no second speculative injection, fault just ignored
+            cbase + 8, mbase + 2 # speculatively execute the last instruction and rollback
+            # terminate after rollback
+            ])) 
+        print(model.tracer.get_contract_trace_full())
```
Comment (OleksiiOleksenko) [inline src/x86/tests/unit_model.py:512]: Leftover print

File: src/x86/tests/unit_model.py [hunk 2]
```diff
@@ -364,6 +467,73 @@ def test_ct_meltdown(self):
...
+        self.assertEqual(ctraces[0], expected_trace)    
+        
+    @unittest.skip("not implemented")
```
Comment (OleksiiOleksenko) [inline src/x86/tests/unit_model.py:515]: Is this test not supposed to be executed?

**Pull request closed without merging**